client/ayon_resolve/plugins/load/load_editorial_package.py

```diff
@@ -32,21 +36,126 @@ def load(self, context, name, namespace, data):
 
         project = lib.get_current_project()
         media_pool = project.GetMediaPool()
+        folder_path = context["folder"]["path"]
 
         # create versioned bin for editorial package
         version_name = context["version"]["name"]
-        bin_name = f"{name}_{version_name}"
-        lib.create_bin(bin_name)
-
+        loaded_bin = lib.create_bin(f"{folder_path}/{name}/{version_name}")
+        loaded_timeline_name = f"{name}_{version_name}_timeline"
         import_options = {
-            "timelineName": "Editorial Package Timeline",
+            "timelineName": loaded_timeline_name,
             "importSourceClips": True,
             "sourceClipsPath": search_folder_path.as_posix(),
         }
 
+        # import timeline from otio file
         timeline = media_pool.ImportTimelineFromFile(files, import_options)
+
+        # get timeline media pool item for metadata update
+        timeline_media_pool_item = None
+        for item in lib.iter_all_media_pool_clips(root=loaded_bin):
+            item_name = item.GetName()
+            if item_name != loaded_timeline_name:
+                continue
+            timeline_media_pool_item = item
+            break
+
+        # Update the metadata
+        if timeline_media_pool_item:
```

Comment:
Shouldn't the timeline media pool item always exist? What am I missing? This should either error if not found... or?

Reply:
improved